llvm_jni/modules/jni.llvm/Module.bal

```diff
@@ -0,0 +1,123 @@
+
+public type TargetTriple string;
+
+function intrinsicNameToId(IntrinsicFunctionName name) returns int {
+    string str_name = "llvm." + name;
+    return llvm_lookup_intrinsic_id(java:fromString(str_name), str_name.length());
+}
+
+public distinct class Module {
+    handle LLVMModule;
+    Context context;
+
+    function init(string moduleName, Context context) {
+        self.context = context;
+        self.LLVMModule = create_llvm_moduleRef(java:fromString(moduleName), context.LLVMContext);
+    }
+
+    function addFunction(string fnName, FunctionType fnType) returns Function {
+        handle llvmFunction = add_function(self.LLVMModule, java:fromString(fnName), toLLVmFunctionType(fnType, self.context));
+        Function fn = new (llvmFunction, fnType, self.context);
+        return fn;
+    }
+
+    public function addFunctionDefn(string name, FunctionType fnType) returns FunctionDefn {
+        return self.addFunction(name, fnType);
+    }
+
+    public function addFunctionDecl(string name, FunctionType fnType) returns FunctionDecl {
+        return self.addFunction(name, fnType);
+    }
+
+    public function printModuleToString() returns string {
+        BytePointer bytePointer = new (module_to_string(self.LLVMModule));
+        return bytePointer.toString();
+    }
+
+    public function printModuleToFile(string fileName) returns io:Error? {
+        byte[] e = [];
+        handle err = checkpanic jarrays:toHandle(e, "byte");
+        _ = module_to_file(self.LLVMModule, java:fromString(fileName), err);
+    }
+
+    public function addGlobal(Type ty, string name) returns PointerValue {
+        return new (llvm_add_global(self.LLVMModule, typeToLLVMType(ty), java:fromString(name)));
+    }
+
+    public function getIntrinsicDeclaration(IntrinsicFunctionName name) returns FunctionDecl {
+        int id = intrinsicNameToId(name);
+        if name is IntegerArithmeticIntrinsicName {
+            FunctionType fnType = {returnType: structType(["i64", "i1"]), paramTypes: ["i64", "i64"]};
+            PointerPointer paramTypes = PointerPointerFromTypes(fnType.paramTypes);
+            return new (llvm_get_intrinsic_declaration(self.LLVMModule, id, paramTypes.jObject, 2), fnType, self.context);
+        }
+        if name is "ptrmask.p0i8.i64" {
+            FunctionType fnType = {returnType: pointerType("i8"), paramTypes: [pointerType("i8"), "i64"]};
+            PointerPointer paramTypes = PointerPointerFromTypes(fnType.paramTypes);
+            return new (llvm_get_intrinsic_declaration(self.LLVMModule, id, paramTypes.jObject, 2), fnType, self.context);
+
+        }
+        panic error(string `${<string>name} not implemented`);
+    }
+
+    public function setTarget(TargetTriple targetTriple) {
+        llvm_set_target(self.LLVMModule, java:fromString(targetTriple));
+    }
+}
+
+function create_llvm_moduleRef(handle moduleId, handle context) returns handle = @java:Method {
```

Comment:
Why are we using names_with_underscores here? We should stick to consistent camelCase convention for Ballerina code.

Reply:
`bal bindgen` generate two functions to each java function. External function in snake case and public function that is exposed to outside in camelCase. I thought it is best if we fallow the same convention specially given in most cases there is a public function with a similar name for every external function we have.

Reply:
How is `bindgen` relevant? Are we using `bindgen`?

Reply:
The naming convention should produce a Ballerina name from the  Java class/method name. I would suggest LLVMFooBarBaz turns into jFooBarBaz.